chain/network/src/peer_manager.rs

```diff
@@ -275,7 +275,15 @@ impl PeerManagerActor {
 
             // TODO: check if peer is banned or known based on IP address and port.
             Peer::add_stream(
-                FramedRead::new(read, Codec::new()).map(|x| x.expect("Decoder error")),
+                FramedRead::new(read, Codec::new())
+                    .take_while(|x| match x {
+                        Ok(_) => future::ready(true),
+                        Err(e) => {
+                            error!(target: "network", "Peer stream error: {:?}", e);
```

Comment:
If the peer stream closes I don't think there is a need to log in `error!`

Reply:
We should at least log in `warn` since this is not expected behavior. I agree that this is not an error in this node, but about node in the other side.